lizmap/modules/lizmap/classes/qgisForm.class.php

```diff
@@ -459,6 +514,109 @@ public function setFormDataFromFields($feature)
         return $form;
     }
 
+    public function check($feature=null)
+    {
+        $form = $this->form;
+
+        $dataFields = $this->dbFieldsInfo->dataFields;
+        $geometryColumn = $this->dbFieldsInfo->geometryColumn;
+
+        // Jelix check
+        $check = $form->check();
+
+        // Geom check
+        $modifyGeometry = $this->layer->getEditionCapabilities()->capabilities->modifyGeometry;
+        if (strtolower($modifyGeometry) == 'true' && $form->getData($geometryColumn) == '') {
+            $check = false;
+            $form->setErrorOn($geometryColumn, jLocale::get('view~edition.message.error.no.geometry'));
+        }
+
+        // Get values and form fields
+        $values = array();
+        $formFields = array();
+        foreach ($dataFields as $fieldName => $prop) {
+            $values[$fieldName] = null;
+            $formFields[] = $fieldName;
+        }
+        if ($feature) {
+            $values = $this->layer->getDbFieldValues($feature);
+        }
+
+        // Get list of fields diplayed in form
+        // can be an empty list
+        $attributeEditorForm = $this->getAttributesEditorForm();
+        if ($attributeEditorForm) {
+            $formFields = $attributeEditorForm->getFields();
+        }
+
+        // Get values from form and get expressions
+        $constraintExpressions = array();
+        foreach ($formFields as $fieldName) {
+            $jCtrl = $form->getControl($fieldName);
+            // Field not in form
+            if ($jCtrl === null) {
+                continue;
+            }
+            // Control is an upload control
+            if ($jCtrl instanceof jFormsControlUpload) {
+                $values[$fieldName] = $this->processUploadedFile($form, $fieldName, $cnx);
```

Comment:
@rldhont It is not possible to upload a file in master. In Montpellier project, point of interest layer, you can try to. It seems because `$this->processUploadedFile()` is called twice, once here in `$qgisForm->check()` and once in `$qgisForm->saveToDb()` in edition.classic.php. For some reasons it fails the second time. I think your check() methos do to much things.